packages/path_provider/example/lib/main.dart

```diff
@@ -140,7 +171,39 @@ class _MyHomePageState extends State<MyHomePage> {
               ),
             ),
             FutureBuilder<Directory>(
-                future: _externalDocumentsDirectory, builder: _buildDirectory)
+                future: _externalDocumentsDirectory, builder: _buildDirectory),
+            Column(children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: RaisedButton(
+                  child: Text(
+                      '${Platform.isIOS ? "External directories are unavailable " "on iOS" : "Get External Storage Directories"}'),
+                  onPressed: Platform.isIOS
+                      ? null
+                      : () {
+                          _requestExternalStorageDirectories(
+                            StorageDirectory.music,
+                          );
+                        },
+                ),
+              ),
+            ]),
+            FutureBuilder<List<Directory>>(
+                future: _externalStorageDirectories,
+                builder: _buildDirectories),
+            Column(children: <Widget>[
```

Comment:
@cyanglaz I have followed the style in the current example code, but think the indentation could be shown better with a few additional "," here & there. Perhaps better to be added in a future PR?